cmd/query/app/querysvc/query_service_v2.go

```diff
@@ -0,0 +1,141 @@
+// // Copyright (c) 2019 The Jaeger Authors.
+// // SPDX-License-Identifier: Apache-2.0
+
+package querysvc
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/model/adjuster"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/depstore"
+	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
+)
+
+// TODO: Remove query_service.go and rename query_service_v2.go
+// to query_service.go once all components have been migrated to
+// operate on the OTEL data model.
+var errNoArchiveSpanStorageV2 = errors.New("archive span storage was not configured")
+
+const (
+	defaultMaxClockSkewAdjustV2 = time.Second
+)
+
+// QueryServiceOptions has optional members of QueryService
+type QueryServiceOptionsV2 struct {
+	ArchiveTraceReader tracestore.Reader
+	ArchiveTraceWriter tracestore.Writer
+	Adjuster           adjuster.Adjuster
+}
```

Comment:
@yurishkuro is there any reason this can't just be part of the `QueryServiceV2` struct?

Reply:
it's a question of now the constructor organized. It has some mandatory arguments and some optional arguments. I don't like the functional option pattern for optional arguments - too much typing overhead with no additional benefits over just passing an Options struct. So the constructor could theoretically be just `New(Settings)`, but that allows required args not to be set. It could be `New(...all args...)` which creates very long signature with unchecked semantics (you could mistaken main reader and archive reader positions, for example). In v1 case it's `New(reader, deps, options)` which clearly indicates required / optional args.